WebStorm IDE Instructions Not Working - Incorrect relative path prefix inserted in front of absolute paths causing Exit Code 1

[IDE Instructions](http://zinserjan.github.io/mocha-webpack/docs/guides/ide-integration.html) for WebStorm 2017 do not seem to work, or I am configuring something incorrectly.

I have both VS Code, and npm command line working fine in bash shell, but need WebStorm due to better (live) debugger integration that currently possible in VS Code.

When running a debug session in WebStorm, it appears mocha-webpack is prefixing a relative path (`./../../..`) in front of absolute paths when trying to import modules.

ERROR OUTPUT:
```
WEBPACK  Compiling...

WEBPACK  Failed to compile with 1 error(s)

Error in ./~/mocha-webpack/lib/entry.js

Module not found: './../../../Users/n8/dev/project_workspace/test/model-tests/Widgets.test.ts' in '/Users/n8/dev/project_workspace/node_modules/mocha-webpack/lib'


Process finished with exit code 1
```

--- a/src/runner/TestRunner.js
+++ b/src/runner/TestRunner.js
@@ -5,6 +5,7 @@ import _ from 'lodash';
 import chokidar from 'chokidar';
 
 import { glob } from '../util/glob';
+import { ensureAbsolutePath } from '../util/paths';
 import createCompiler from '../webpack/compiler/createCompiler';
 import createWatchCompiler from '../webpack/compiler/createWatchCompiler';
 import registerInMemoryCompiler from '../webpack/compiler/registerInMemoryCompiler';
@@ -244,7 +245,7 @@ export default class TestRunner extends EventEmitter {
 
     const entryConfig = new EntryConfig();
     files
-      .map((f) => path.join(this.options.cwd, f))
+      .map((f) => ensureAbsolutePath(f, this.options.cwd))
       .forEach((f) => entryConfig.addFile(f));
 
     const includeLoaderQuery = {

--- a/src/util/paths.js
+++ b/src/util/paths.js
@@ -1,0 +1,8 @@
+// @flow
+import { join, posix, win32 } from 'path';
+
+// eslint-disable-next-line import/prefer-default-export
+export function ensureAbsolutePath(path: string, basePath: string) {
+  return posix.isAbsolute(path) || win32.isAbsolute(path) ? path : join(basePath, path);
+}
+

--- a/test/integration/cli/entry.test.js
+++ b/test/integration/cli/entry.test.js
@@ -157,6 +157,26 @@ describe('cli - entry', function () {
     });
   });
 
+  context('entry with absolute paths', function () {
+    before(function () {
+      this.passingTest = path.join(process.cwd(), fixtureDirTmp, 'passing-test.js');
+      createTest(this.passingTest, true);
+    });
+
+    it('runs test with absolute entry', function (done) {
+      exec(`node ${binPath} "${this.passingTest}"`, (err, stdout) => {
+        assert.isNull(err);
+        assert.include(stdout, this.passingTest);
+        assert.include(stdout, '1 passing');
+        done();
+      });
+    });
+
+    after(function () {
+      return del([this.passingTest]);
+    });
+  });
+
   context('glob pattern as option', function () {
     const testFiles = _.range(1, 30).map((x) => {
       if (parseInt(x / 10, 10) === 0) {

--- a/test/integration/cli/entry.test.js
+++ b/test/integration/cli/entry.test.js
@@ -10,11 +10,12 @@ import { exec } from 'child_process';
 import anymatch from 'anymatch';
 import normalizePath from 'normalize-path';
 
+const escapePath = (p) => p.replace(/\\/gm, '\\\\');
 
 function createTest(filePath, passing) {
   const content = `
     var assert = require('assert');
-    describe('${filePath}', function () {
+    describe('${escapePath(filePath)}', function () {
       it('runs test', function () {
         assert.ok(${passing});
       });
@@ -26,7 +27,7 @@ function createTest(filePath, passing) {
 function createCorruptedTest(filePath) {
   const content = `
     var assert = require('assert');
-    describe('${filePath}', function () {
+    describe('${escapePath(filePath)}', function () {
       it('runs test', function () {
         assert.ok(false);
     });
@@ -37,8 +38,8 @@ function createCorruptedTest(filePath) {
 function createRuntimeErrorTest(filePath, passing) {
   const content = `
     var assert = require('assert');
-    throw new Error('error in ${filePath}');
-    describe('${filePath}', function () {
+    throw new Error('error in ${escapePath(filePath)}');
+    describe('${escapePath(filePath)}', function () {
       it('runs test', function () {
         assert.ok(${passing});
       });

